pilot/pkg/security/authn/utils/utils.go

```diff
@@ -110,7 +110,7 @@ func BuildInboundFilterChain(mTLSMode model.MutualTLSMode, sdsUdsPath string, no
 func GetSAN(ns string, identity string) string {
 
 	if ns != "" {
-		return spiffe.MustGenSpiffeURI(ns, identity)
+		return spiffe.MustGenSpiffeURI(spiffe.GetTrustDomain(), ns, identity)
```

Comment:
Please stop using the spiffee singleton ( GetTrustDomain ) - instead use MeshConfig.TrustDomain. 

I don't know who added the 4th way to configure ( a static global in spiffee - besides the MeshConfig, cmd line flag and env variable ). But from the code it doesn't look that the global in spiffee is updated when mesh config changes - and there is no need for that.

Reply:
Also please make sure this works with aliases - where is this method called ? I tried to set the aliases, and it doesn't appear to generate the proper result - I suspect this method should return an array of strings.

Reply:
removed, since no need...